projects/packages/forms/src/contact-form/class-contact-form-field.php

```diff
@@ -848,6 +870,78 @@ public function render_telephone_field( $id, $label, $value, $class, $required,
 		return $field;
 	}
 
+	/**
+	 * Return the HTML for the telephone field.
+	 *
+	 * @param int    $id - the ID.
+	 * @param string $label - the label.
+	 * @param string $value - the value of the field.
+	 * @param string $class - the field class.
+	 * @param bool   $required - if the field is marked as required.
+	 * @param string $required_field_text - the text in the required text field.
+	 * @param string $placeholder - the field placeholder content.
+	 *
+	 * @return string HTML
+	 */
+	public function render_phone_field( $id, $label, $value, $class, $required, $required_field_text, $placeholder ) {
+		$this->set_invalid_message( 'phone', __( 'Please enter a valid phone number', 'jetpack-forms' ) );
+		$label = $this->render_label( 'phone', $id, $label, $required, $required_field_text );
+
+		// since now the telephone field can include a country selector, we need to handle the rendering of the field differently
+		$styles = '';
+
+		if ( ! empty( $this->field_styles ) ) {
+			$styles .= sprintf( 'style="%s" ', esc_attr( $this->field_styles ) );
+		}
+
+		if ( ! is_string( $value ) ) {
+			$value = '';
+		}
+
+		$id               = esc_attr( $id );
+		$country_selector = '';
+
+		// Country selector: select element wrapped in div.jetpack-field__input-prefix
+		if ( $this->get_attribute( 'showcountryselector' ) ) {
+			$country_data    = $this->get_attribute( 'countrydata' );
+			$default_country = $value ? $value : $this->get_attribute( 'default' );
+
+			if ( ! empty( $country_data ) ) {
+				$country_options = array_map(
+					function ( $country ) use ( $default_country ) {
+						return sprintf( '<option value="%s" %s>%s</option>', $country['value'], selected( $country['value'], $default_country, false ), $country['label'] );
+					},
+					$country_data
+				);
+
+				$country_selector = sprintf( '<div class="jetpack-field__input-prefix"><select %s>%s</select></div>', $class, implode( '\n', $country_options ) );
+			}
+		}
+
+		$input  = $country_selector;
+		$input .= "<input
+			type='tel'
+			name='" . $id . "'
+			id='" . $id . "'
+			value=''
+
+			data-wp-bind--aria-invalid='state.fieldHasErrors'
+			data-wp-bind--value='state.getFieldValue'
+			aria-errormessage='" . $id . '-' . "tel-error-message'
+			data-wp-on--input='actions.onFieldChange'
+			data-wp-on--blur='actions.onFieldBlur'
+			data-wp-class--has-value='state.hasFieldValue'
+
+			" . $class . $placeholder . '
+			' . ( $required ? "required='true' aria-required='true' " : '' ) .
+			" />\n ";
+
+		$wrapped_input = sprintf( '<div class="jetpack-field__input-phone-wrapper %s" %s>%s</div>', $this->get_attribute( 'stylevariationclasses' ), $styles, $input );
+
+		$field = $label . $wrapped_input . $this->get_error_div( $id, 'tel' );
```

Comment:
Should this be 'tel' ?

Reply:
Likely this is messing with the validation, something I'll be looking next. For this iteration I think I should change it to `tel` as you mention, but at some point the validation for phone numbers can become really complex and dependent on a more robust library. TBD.